From 2ffdbc9bd2c12ea37c6c6cd340df26be0e0aaf96 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Tue, 6 Dec 2016 09:41:04 +0800
Subject: [PATCH 1342/5242] MLK-13570-4 usb: chipidea: usbmisc_imx: add
 imx7ulp entry

commit  bee188567a174d7747107823ec729180aeb25be2 from
https://source.codeaurora.org/external/imx/linux-imx.git

imx7ulp non core register mapping is similar with imx7d, and the
initialization is the same, but lacks of USB charger detection support.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/usb/usbmisc-imx.txt        |    1 +
 drivers/usb/chipidea/usbmisc_imx.c                 |    9 +++++++++
 2 files changed, 10 insertions(+)

diff --git a/Documentation/devicetree/bindings/usb/usbmisc-imx.txt b/Documentation/devicetree/bindings/usb/usbmisc-imx.txt
index 0ff9017..a7fafac 100644
--- a/Documentation/devicetree/bindings/usb/usbmisc-imx.txt
+++ b/Documentation/devicetree/bindings/usb/usbmisc-imx.txt
@@ -8,6 +8,7 @@ Required properties:
 	"fsl,imx6sx-usbmisc" for imx6sx
 	"fsl,imx7d-usbmisc" for imx7d
 	"fsl,imx6ul-usbmisc" for imx6ul
+	"fsl,imx7ulp-usbmisc" for imx7ulp
 - reg: Should contain registers location and length
 
 Examples:
diff --git a/drivers/usb/chipidea/usbmisc_imx.c b/drivers/usb/chipidea/usbmisc_imx.c
index e052aae..f8a1102 100644
--- a/drivers/usb/chipidea/usbmisc_imx.c
+++ b/drivers/usb/chipidea/usbmisc_imx.c
@@ -1026,6 +1026,11 @@ static int usbmisc_term_select_override(struct imx_usbmisc_data *data,
 	.term_select_override = usbmisc_term_select_override,
 };
 
+static const struct usbmisc_ops imx7ulp_usbmisc_ops = {
+	.init = usbmisc_imx7d_init,
+	.set_wakeup = usbmisc_imx7d_set_wakeup,
+};
+
 static inline bool is_imx53_usbmisc(struct imx_usbmisc_data *data)
 {
 	struct imx_usbmisc *usbmisc = dev_get_drvdata(data->dev);
@@ -1225,6 +1230,10 @@ int imx_usbmisc_term_select_override(struct imx_usbmisc_data *data,
 		.compatible = "fsl,imx7d-usbmisc",
 		.data = &imx7d_usbmisc_ops,
 	},
+	{
+		.compatible = "fsl,imx7ulp-usbmisc",
+		.data = &imx7ulp_usbmisc_ops,
+	},
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, usbmisc_imx_dt_ids);
-- 
1.7.9.5

